components/infobox/wikis/tft/infobox_league_custom.lua

```diff
@@ -0,0 +1,108 @@
+---
+-- @Liquipedia
+-- wiki=tft
+-- page=Module:Infobox/League/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Variables = require('Module:Variables')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local League = Lua.import('Module:Infobox/League', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+
+local _args
+
+local CustomLeague = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local GAME_MODE = mw.loadData('Module:GameMode')
+local RIOT_ICON = '[[File:Riot Games Tier Icon.png|x12px|link=Riot Games|Tournament supported by Riot Games]]'
+
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_args = league.args
+
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+	league.liquipediaTierHighlighted = CustomLeague.liquipediaTierHighlighted
+	league.appendLiquipediatierDisplay = CustomLeague.appendLiquipediatierDisplay
+
+	return league:createInfobox(frame)
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	local args = _args
+	table.insert(widgets, Cell{
+		name = 'Teams',
+		content = {args.team_number}
+	})
+	table.insert(widgets, Cell{
+		name = 'Players',
+		content = {args.participants_number}
+	})
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'gamesettings' then
+		table.insert(widgets, Cell{
+			name = 'Patch',
+			content = {CustomLeague:_createPatchCell(_args)}
+		})
+		table.insert(widgets, Cell{
+			name = 'Game Mode',
+			content = {CustomLeague._getGameMode()}
+		})
+	end
+
+	return widgets
+end
+
+function CustomLeague:liquipediaTierHighlighted(args)
+	return Logic.readBool(args['riot-sponsored'])
+end
+
+function CustomLeague:appendLiquipediatierDisplay()
+	if Logic.readBool(_args['riot-sponsored']) then
+		return ' ' .. RIOT_ICON
+	end
+	return ''
+end
+
+function CustomLeague:defineCustomPageVariables()
+	Variables.varDefine('tournament_mode', string.lower(_args.mode or ''))
```

Comment:
this is imo not a good idea
it just stores the input plain (lowercased)
this will lead to
- invalid values getting set/stored
- different values getting set/stored for the same display (e.g. `solo` vs `solos`)

in my opinion the `GameMode` should be adjusted to only return the standardized value (or just kick that module and do it directly in infobox league)
that way you can build the display, category and storage value via it
the only current usecase for that module is inside infobox league anyways:
https://liquipedia.net/tft/index.php?title=Special:Search&profile=all&search=insource%3AGameMode&fulltext=1